SDL/TrackCandidate.cu

```diff
@@ -79,6 +79,23 @@ __device__ void SDL::addTrackCandidateToMemory(struct trackCandidates& trackCand
     trackCandidatesInGPU.objectIndices[2 * trackCandidateIndex + 1] = outerTrackletIndex;
 }
 
+__device__ void SDL::addTrackCandidateToMemory(struct trackCandidates& trackCandidatesInGPU, short trackCandidateType, unsigned int innerTrackletIndex, unsigned int outerTrackletIndex, unsigned int trackCandidateIndex, unsigned int directObjectIndex, uint4 hitIndices)
+{
+    trackCandidatesInGPU.trackCandidateType[trackCandidateIndex] = trackCandidateType;
+    trackCandidatesInGPU.directObjectIndices[trackCandidateIndex] = directObjectIndex;
+    trackCandidatesInGPU.objectIndices[2 * trackCandidateIndex] = innerTrackletIndex;
+    trackCandidatesInGPU.objectIndices[2 * trackCandidateIndex + 1] = outerTrackletIndex;
+
+    // if it is pLS add the hit indices
+    if (trackCandidateType == 8)
+    {
+        trackCandidatesInGPU.hitIndices[14 * trackCandidateIndex + 0] = hitIndices.x; // x z y w is the correct order due to how it was added in the "addPixelSegmentToMemory"
```

Comment:
@sgnoohc I missed this the first time I was reading the code and it is relevant for my PR; this comment (and the code implementation) seems to be in contradiction with [the lines in addPixelSegmentToMemory](https://github.com/SegmentLinking/TrackLooper/blob/ce2f36b24a572c78e6aa9b7a17e61d8d0fc7e10a/SDL/Segment.cu#L304-L307). Am I missing something?

Reply:
You must follow one more level up ;)
https://github.com/SegmentLinking/TrackLooper/blob/ce2f36b24a572c78e6aa9b7a17e61d8d0fc7e10a/SDL/Event.cu#L669-L672
There you can see that the order is anchor anchor outer outer. whcy makes it 0 2 1 3 order.
We truly have a physicists' code!

Philip

Reply:
Thanks, I understand now. Let me follow up in the dedicated issue you had created (sorry, I had missed that).